Sketch.java

```diff
@@ -17,20 +36,143 @@ public void settings() {
    */
   public void setup() {
     background(210, 255, 173);
+    // Changes the height of each circle in the array
+    for (int i = 0; i < circleY.length; i++) {
+      circleY[i] = random(height);
+    }
+    for (int i = 0; i < circleX.length; i++) {
+      circleX[i] = random(width);
+    }
   }
 
   /**
    * Called repeatedly, anything drawn to the screen goes here
    */
   public void draw() {
-	  
-	// sample code, delete this stuff
-    stroke(128);
-    line(150, 25, 270, 350);  
-
-    stroke(255);
-    line(50, 125, 70, 50);  
+	  // Background
+    background(50);
+    // Creating the circles that fall down
+    for (int i = 0; i < circleY.length; i++) {
+      if(circleHidden[i] == false){
+      fill(255,255,255);
+      ellipse(circleX[i], circleY[i], 25, 25);
+      }
+    // Collision detection
+    if(dist((float)PlayerX,(float)PlayerY,circleX[i],circleY[i]) < circleRadius + playerRadius){
+      intLives = intLives - 1;
+      circleY[i] = 0;
```

Comment:
need to include circleHidden logic here.  based on what you have, you are able to collide with hidden circles.